src/go/k8s/cmd/configurator/main.go

```diff
@@ -40,20 +40,24 @@ const (
 	externalConnectivityEnvVar          = "EXTERNAL_CONNECTIVITY"
 	externalConnectivitySubDomainEnvVar = "EXTERNAL_CONNECTIVITY_SUBDOMAIN"
 	hostPortEnvVar                      = "HOST_PORT"
+	proxyExternalConnectivityEnvVar     = "PROXY_EXTERNAL_CONNECTIVITY"
+	proxyHostPortEnvVar                 = "PROXY_HOST_PORT"
 )
 
 type brokerID int
 
 type configuratorConfig struct {
-	hostName             string
-	svcFQDN              string
-	configSourceDir      string
-	configDestination    string
-	nodeName             string
-	subdomain            string
-	externalConnectivity bool
-	redpandaRPCPort      int
-	hostPort             int
+	hostName                  string
+	svcFQDN                   string
+	configSourceDir           string
+	configDestination         string
+	nodeName                  string
+	subdomain                 string
+	externalConnectivity      bool
+	redpandaRPCPort           int
+	hostPort                  int
+	proxyExternalConnectivity bool
+	proxyHostPort             int
```

Comment:
if we pull the nodeport service from here, is it also problematic as pulling the CRD? (just asking since you said you tried it :))

Reply:
It works fine for core resources and CRDs. We need a couple of additions in the cluster role the operator creates.

I have a working branch with this approach if we ever need to go that route.